### PR TITLE
Reintroduce and deprecate API removed in 1.8.0.

### PR DESCRIPTION
**Pull request by io-german**
- Reintroduce protected connection field to InfileObjectLoader class.
- Deprecate all protected fields in InfileObjectLoader and SingleInfileObjectLoader classes. Introduce accessors/mutators for these fields. Access level modifiers for them is the same as for the fields.
- Extend new StatementCallback interface from old one.
- Minor changes to the Javadoc of ConnectionBasedStatementExecutor and DataSourceBasedStatementExecutor.

**There is still one difference in API remaining**. Interface ```StatementExecutor``` was moved out of ```JdbcUtil```. I tried to extend new interface from the old one to achieve backward compatibility. This approach result in failing build because of deprecation warning that can't be suppressed due to known bug in javac that was fixed only in Java 8 (http://bugs.java.com/view_bug.do?bug_id=6480588).

### CODE REVIEW

**Comment by corcorb** [comment]
On first look through this looks good. I'll try testing it out later today — will hopefully merge & release by EOD.

File: release_notes.txt [hunk 1]
```diff
@@ -1,3 +1,6 @@
+New in 1.8.1
+    Reintroduced and deprecated the API removed in 1.8.0 release
+
 New in 1.8.0
```
Comment (corcorb) [inline release_notes.txt]: Let's make a note here that this version broke backwards compatibility and should not be used.

File: src/main/java/com/opower/persistence/jpile/loader/SingleInfileObjectLoader.java [hunk 2]
```diff
@@ -83,6 +247,7 @@ public void convertToInfileRow(E entity, InfileRow infileRow) {
                 if (this.persistenceAnnotationInspector.hasTableAnnotation(object.getClass())) {
                     Long id = (Long) getIdValue(this.persistenceAnnotationInspector, object);
                     Preconditions.checkState(id != null, "@Id for [%s] is null", object);
+                    /* TODO: this statement is always true */
```
Comment (corcorb) [inline src/main/java/com/opower/persistence/jpile/loader/SingleInfileObjectLoader.java]: This was added to make FindBugs' null-pointer analysis happy, but I can no longer reproduce the error. Let's just remove the assertion.

Comment (io-german) [inline src/main/java/com/opower/persistence/jpile/loader/SingleInfileObjectLoader.java]: Done

File: release_notes.txt (hunk 1 above)
Comment (io-german) [inline release_notes.txt]: Done

**Comment by corcorb** [comment]
I've verified that this version maintains compatibility with `libs-db` and `inbound`. Merging now, and will start release process. Thanks @io-german !

**Pull request merged**